### PR TITLE
Hollow Knight: Add additional DeathLink option and add ExtraPlatforms option.

### PR DESCRIPTION
**Pull request by ThePhar**
## What is this fixing or adding?
https://github.com/ArchipelagoMW-HollowKnight/Archipelago.HollowKnight/issues/157 - Splits DeathLink options up to allow more configuration of behavior on death without needing to program all possible combinations in existing DeathLink option.

https://github.com/ArchipelagoMW-HollowKnight/Archipelago.HollowKnight/issues/58 - Also adds an "ExtraPlatforms" option to create additional platforms in Hallownest to make moving around easier.

@BadMagic100 - Leaving in Draft until client-side can be updated and verified they work with these changes.

## How was this tested?
No logic changes were made, but verified output on WebHost. Awaiting client-side tests before pulling out of Draft state.

## If this makes graphical changes, please attach screenshots.
N/A

### CODE REVIEW

File: worlds/hk/Options.py [hunk 1]
```diff
@@ -402,22 +402,30 @@ class WhitePalace(Choice):
...
+    option_vanilla = 0
     option_shadeless = 1
-    option_vanilla = 2
-    option_shade = 3
+    option_shade = 2
+    default = 2
+
+
+class DeathLinkBreaksFragileCharms(Toggle):
+    """Sets if fragile charms break when you are killed by a DeathLink. All other deaths still break fragile charms.
```
Comment (BadMagic100) [inline worlds/hk/Options.py:425]: As noted by folks on discord it's probably slightly more accurate to say something like "fragile charms will still be broken by your own deaths as usual" to cover for cases where your own deaths actually will not break the charm

File: worlds/hk/Options.py [hunk 2]
```diff
@@ -402,22 +402,30 @@ class WhitePalace(Choice):
...
-    When enabled, choose how incoming deathlinks are handled:
-    vanilla: DeathLink kills you and is just like any other death.  RIP your previous shade and geo.
-    shadeless: DeathLink kills you, but no shade spawns and no geo is lost.  Your previous shade, if any, is untouched.
-    shade: DeathLink functions like a normal death if you do not already have a shade, shadeless otherwise.
+class ExtraPlatforms(DefaultOnToggle):
+    """Places additional platforms to make traveling throughout Hallownest more convenient."""
+
+
+class DeathLinkShade(Choice):
+    """Sets whether to create a shade when you are killed by a DeathLink and how to handle your existing shade, if any.
```
Comment (BadMagic100) [inline worlds/hk/Options.py:410]: Similar to the fragile charms setting a caveat that your own deaths will create a shade and remove any existing shade as normal would be good

**Review by BadMagic100 — CHANGES_REQUESTED** [review]
Will work on the client side changes this weekend and update here once they are merged

**Comment by ThePhar** [comment]
Personally, I don't count those as actual deaths anyway if it doesn't drop a shade. E.g. dream fights.

**Comment by BadMagic100** [comment]
A deathlink is still sent even in cases you normally wouldn't leave a shade or end steel soul. In any case I think the comment regarding shades probably still stands.

Maybe the better way to phrase it is to put somewhere that "the penalties of your own deaths inflicted on you are unmodified," either on the deathlink toggle or with respect to each optoo

**Comment by BadMagic100** [comment]
(closing this was unintentional, the mobile UI had it out for me)

**Comment by ThePhar** [comment]
Made some more verbiage tweaks.

**Review by BadMagic100 — APPROVED** [review]

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**